[Improve][Zeta] Fix engine e2e error

EricJoy2048 opened the pull request:
1. Make all Asynchronous operation use specified thread pool, and not allow to use the thread pool created by jdk (`ForkJoinPool.commonPool()`)
2. Update deleteCheckpoint and support batch delete checkpoint.
3. Enable zeta e2e test.
4. Reduce the queue size(100000 update to 2048) used between source task and sink task, because it may cause oom.
5. Fix the zeta scheduler bug.
6. Fix the zeta checkpoint bug.

EricJoy2048 committed:
enable zeta e2e

EricJoy2048 committed:
Merge remote-tracking branch 'apache/dev' into 62_enable_zeta_e2e

EricJoy2048 committed:
Merge remote-tracking branch 'apache/dev' into 62_enable_zeta_e2e

EricJoy2048 committed:
fix job restore bug

EricJoy2048 committed:
merge from dev

EricJoy2048 committed:
fix bug

EricJoy2048 committed:
Merge remote-tracking branch 'apache/dev' into 62_enable_zeta_e2e

EricJoy2048 committed:
fix zeta e2e

EricJoy2048 committed:
tmp commit

ic4y committed:
add interrupt

ic4y committed:
add interrupt

ic4y committed:
add print taskExecutionServer Runtime Info

ic4y committed:
fix checkstyle

ic4y committed:
add license header

ic4y committed:
fix err

ic4y committed:
fix checkstyle

EricJoy2048 committed:
fix ci bug

EricJoy2048 committed:
format code use spotless

EricJoy2048 committed:
merge from dev

EricJoy2048 committed:
merge from dev

EricJoy2048 committed:
merge from dev

EricJoy2048 committed:
format code

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Tyrantlucifer <tyrantlucifer@apache.org>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Tyrantlucifer <tyrantlucifer@apache.org>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Tyrantlucifer <tyrantlucifer@apache.org>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Tyrantlucifer <tyrantlucifer@apache.org>

EricJoy2048 committed:
Update seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeSourceSplitEnumerator.java

Co-authored-by: Tyrantlucifer <tyrantlucifer@apache.org>

EricJoy2048 committed:
Update seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeSourceSplitEnumerator.java

Co-authored-by: Tyrantlucifer <tyrantlucifer@apache.org>

EricJoy2048 committed:
Update seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeSourceSplitEnumerator.java

Co-authored-by: Tyrantlucifer <tyrantlucifer@apache.org>

EricJoy2048 committed:
fix cdc oom

EricJoy2048 committed:
fix review comment

EricJoy2048 committed:
Merge branch '62_enable_zeta_e2e' into 70_disable_cdc_cache

EricJoy2048 committed:
fix client timeout and then job cancel by client

EricJoy2048 committed:
fix cdc oom

EricJoy2048 committed:
fix ci error

EricJoy2048 committed:
reduce the queue size

EricJoy2048 committed:
improve log output

EricJoy2048 committed:
disable checkpoint serialize test

EricJoy2048 committed:
improve thread pool use

EricJoy2048 committed:
fix ut error

EricJoy2048 committed:
reduce cpu use

EricJoy2048 committed:
tmp

EricJoy2048 committed:
spotless format

EricJoy2048 committed:
Merge branch '62_enable_zeta_e2e' into 70_disable_cdc_cache

EricJoy2048 committed:
Merge branch '70_disable_cdc_cache' into 71_improve_cpu_load

EricJoy2048 committed:
fix test error

EricJoy2048 committed:
update log level

EricJoy2048 committed:
fix delete checkpoint bug

EricJoy2048 committed:
spotless format

EricJoy2048 committed:
revert seatunnel.sh

EricJoy2048 committed:
add database name and table name to error log

EricJoy2048 committed:
improve log ouput

liugddx commented on config/jvm_options:
```diff
@@ -102,4 +102,7 @@
+8:-XX:GCLogFileSize=64m
+
+-XX:+HeapDumpOnOutOfMemoryError
+-XX:HeapDumpPath=/data/whaletunnel/
```
```suggestion
-XX:HeapDumpPath=/data/seatunnel/
```

EricJoy2048 replied:
done

liugddx commented on seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/IncrementalSource.java:
```diff
@@ -134,7 +134,7 @@ public SourceReader<T, SourceSplitBase> createReader(SourceReader.Context reader
         C sourceConfig = configFactory.create(readerContext.getIndexOfSubtask());
         BlockingQueue<RecordsWithSplitIds<SourceRecords>> elementsQueue =
-                new LinkedBlockingQueue<>(1024);
+                new LinkedBlockingQueue<>(2);
```
Why is 2

EricJoy2048 replied:
1024 is too big and may cause oom.

liugddx commented on seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/reader/SourceReaderBase.java:
```diff
@@ -158,7 +158,12 @@ private RecordsWithSplitIds<E> getNextFetch(Collector<T> output) {
+                log.trace("Current fetch is finished.");
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```
Unified exception?

EricJoy2048 replied:
done

liugddx commented on seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlUtils.java:
```diff
@@ -339,7 +340,12 @@ private static PreparedStatement initStatement(JdbcConnection jdbc, String sql,
+                connection.prepareStatement(
+                        sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+        if (fetchSize == 0) {
+            fetchSize = 1024;
```
```suggestion
            fetchSize = Integer.MIN_VALUE;
```

liugddx replied:
https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-implementation-notes.html

EricJoy2048 replied:
Thanks , done

liugddx commented on seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java:
```diff
@@ -810,56 +830,77 @@ public void testStreamJobRestoreInAllNodeDown()
 
             node2 = SeaTunnelServerStarter.createHazelcastInstance(seaTunnelConfig);
 
+            System.out.println(
+                    "==========================================All node is start, begin check node size ========================================");
```
```suggestion
            log.info(
                    "==========================================All node is start, begin check node size ========================================");
```

liugddx commented on seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java:
```diff
@@ -242,8 +243,8 @@ public PipelineState getCheckpoint(String jobId, String pipelineId, String check
         if (fileNames.isEmpty()) {
-            throw new CheckpointStorageException(
-                    "No checkpoint found for job, job id is: " + jobId);
+            log.info("No checkpoint found for job, job id is: " + jobId);
```
```suggestion
            log.info("No checkpoint found for this job,  the job id is: " + jobId);
```

liugddx commented on seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java:
```diff
@@ -207,8 +208,8 @@ public List<PipelineState> getCheckpointsByJobIdAndPipelineId(String jobId, Stri
         if (fileNames.isEmpty()) {
-            throw new CheckpointStorageException(
-                    "No checkpoint found for job, job id is: " + jobId);
+            log.info("No checkpoint found for job, job id is: " + jobId);
```
```suggestion
            log.info("No checkpoint found for this job, the job id is: " + jobId);
```

liugddx commented on seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java:
```diff
@@ -185,18 +185,19 @@ public PipelineState getLatestCheckpointByJobIdAndPipelineId(String jobId, Strin
         if (latestFileName == null) {
-            throw new CheckpointStorageException(
+            log.info(
                     "No checkpoint found for job, job id is: "
```
```suggestion
                    "No checkpoint found for this job, the job id is: "
```

liugddx commented on seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java:
```diff
@@ -174,7 +174,7 @@ public List<PipelineState> getLatestCheckpoint(String jobId) throws CheckpointSt
 
         if (latestPipelineStates.isEmpty()) {
-            throw new CheckpointStorageException("No checkpoint found for job, job id:{} " + jobId);
+            log.info("No checkpoint found for job, job id:{} " + jobId);
```
```suggestion
            log.info("No checkpoint found for this job,  the job id:{} " + jobId);
```

liugddx commented on seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java:
```diff
@@ -159,8 +159,8 @@ public List<PipelineState> getLatestCheckpoint(String jobId) throws CheckpointSt
         if (fileNames.isEmpty()) {
-            throw new CheckpointStorageException(
-                    "No checkpoint found for job, job id is: " + jobId);
+            log.info("No checkpoint found for job, job id is: " + jobId);
```
```suggestion
            log.info("No checkpoint found for this  job, the job id is: " + jobId);
```

liugddx commented on seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java:
```diff
@@ -135,8 +135,8 @@ public List<PipelineState> getAllCheckpoints(String jobId) throws CheckpointStor
         if (fileNames.isEmpty()) {
-            throw new CheckpointStorageException(
-                    "No checkpoint found for job, job id is: " + jobId);
+            log.info("No checkpoint found for job, job id is: " + jobId);
```
```suggestion
            log.info("No checkpoint found for this job, the job id is: " + jobId);
```

EricJoy2048 committed:
fix cancel job not success bug

EricJoy2048 committed:
spotless format code

EricJoy2048 committed:
Update seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
spotless format code

EricJoy2048 committed:
fix review problem

liugddx commented on seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java:
```diff
@@ -133,6 +133,7 @@ public void testBatchJobRunOkIn2Node() throws ExecutionException, InterruptedExc
                     FileUtils.getFileLineNumberFromDir(testResources.getLeft());
             Assertions.assertEquals(testRowNumber * testParallelism, fileLineNumberFromDir);
             System.out.println(engineClient.getJobMetrics(clientJobProxy.getJobId()));
+            System.out.println("========================clean test resource====================");
```
```suggestion
            log.info("========================clean test resource====================");
```

liugddx commented on seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java:
```diff
@@ -311,6 +312,8 @@ public void testBatchJobRestoreIn2NodeWorkerDown()
                                     Assertions.assertEquals(
                                             2, finalNode.getCluster().getMembers().size()));
 
+            System.out.println(
```
```suggestion
            log.info(
```

liugddx commented on seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java:
```diff
@@ -345,10 +348,12 @@ public void testBatchJobRestoreIn2NodeWorkerDown()
                             });
 
             // shutdown on worker node
+            System.out.println(
```
```suggestion
            log.info(
```

liugddx commented on seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java:
```diff
@@ -559,13 +565,18 @@ public void testBatchJobRestoreIn2NodeMasterDown()
-                                                            objectCompletableFuture.get())));
+                            () -> {
+                                Thread.sleep(2000);
+                                System.out.println(
```
```suggestion
                                log.info(
```

liugddx commented on seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java:
```diff
@@ -810,56 +830,77 @@ public void testStreamJobRestoreInAllNodeDown()
                             () -> {
                                 // Wait job write all rows in file
                                 Thread.sleep(2000);
                                 System.out.println(
```
```suggestion
                                log.info(
```

liugddx commented on seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java:
```diff
@@ -368,18 +367,23 @@ public void testTwoPipelineBatchJobRestoreIn2NodeWorkerDown()
+                            () -> {
+                                // Wait some tasks commit finished
+                                Thread.sleep(2000);
+                                System.out.println(
```
```suggestion
                                log.info(
```

liugddx commented on seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java:
```diff
@@ -592,13 +604,19 @@ public void testTwoPipelineBatchJobRestoreIn2NodeMasterDown()
+                            () -> {
+                                // Wait some tasks commit finished
+                                Thread.sleep(2000);
+                                System.out.println(
```
```suggestion
                                log.info(
```

EricJoy2048 committed:
update log level

Hisoka-X committed:
[Fix] Fix Job Restore Error

Hisoka-X committed:
[Fix] Fix Job Restore Error

EricJoy2048 committed:
Merge branch '71_improve_cpu_load' of https://github.com/EricJoy2048/incubator-seatunnel into 71_improve_cpu_load

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Update seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java

Co-authored-by: Guangdong Liu <804167098@qq.com>

EricJoy2048 committed:
Merge branch '71_improve_cpu_load' of https://github.com/EricJoy2048/incubator-seatunnel into 71_improve_cpu_load

EricJoy2048 committed:
fix checkpoint storage s3 config error

fix s3 checkpoint storage bug

fix checkpoint error can not send to client

fix license problem

fix taskgroup context not found and operation errro bug

fix taskgroup context not found and operation errro bug

TyrantLucifer commented:
Contained in #4256
